examples/Cassie/multibody_sim.cc

```diff
@@ -146,20 +156,80 @@ int do_main(int argc, char* argv[]) {
     // simulator.get_mutable_integrator()->set_target_accuracy(1e-1);
     // simulator.get_mutable_integrator()->set_fixed_step_mode(true);
     simulator.reset_integrator<drake::systems::RungeKutta2Integrator<double>>(
-      FLAGS_dt);
+        FLAGS_dt);
   }
 
   simulator.set_publish_every_time_step(false);
   simulator.set_publish_at_initialization(false);
   simulator.set_target_realtime_rate(FLAGS_target_realtime_rate);
   simulator.Initialize();
-  simulator.AdvanceTo(std::numeric_limits<double>::infinity());
+  simulator.AdvanceTo(FLAGS_end_time);
 
   return 0;
 }
 
-}  // namespace dairlib
+Eigen::VectorXd GetInitialState(const MultibodyPlant<double>& plant) {
```

Comment:
Can you comment a little on the goal of this function, and some of the constraints below? Some specific questions:
- Does this function solve for an equilibrium configuration?
- How do the linear constraints below enforce the loop closure (which would seem nonlinear)?

Reply:
The goal of the function is to make it easier to set the initial configuration for the simulated robot (ie set toe/pelvis positions). I don't think the IK solver necessarily solves for a configuration that is in equilibrium but just from testing, any deviations from equilibrium get damped out quickly (the OSC standing controller reaches equilibrium in 0.2s).

Reply:
What about the loop closure? Also, the toes don't seem to have to be exactly on the ground. Why is this?

Reply:
I'm confused about the loop closure - the loop closure constraints are still implemented in addCassieMultibody in cassie_utils .cc. This function only solves for the initial positions for the simulator and doesn't impose any constraints on the simulator.

The toes start slightly above the ground because the starting configuration isn't perfectly flush with the ground (as in there are 4 contacts). This should be changed, I'll add a change to the IK solver to constrain all 4 contact points to be on the ground.

Reply:
You would probably want the initial state to not have an energy in the loop closure springs (e.g. the loops are properly initialized in a closed configuration). In RBT, this (and finding equlibria) was done via a custom Mathematical Program: https://github.com/DAIRLab/dairlib/blob/master/attic/multibody/multibody_solvers.cc

Reply:
Going to approve, though it's probably a TODO to replicate the fixed point solver from RBT.